nifi-commons/nifi-processor-utilities/src/main/java/org/apache/nifi/processor/util/put/AbstractPutEventProcessor.java

```diff
@@ -119,7 +119,24 @@
             .defaultValue("10 seconds")
             .addValidator(StandardValidators.TIME_PERIOD_VALIDATOR)
             .build();
-
+    public static final PropertyDescriptor OUTGOING_MESSAGE_DELIMITER = new PropertyDescriptor.Builder()
+            .name("Outgoing Message Delimiter")
+            .description("Specifies the delimiter to use when sending messages out over the same TCP stream. The delimiter is appended to each FlowFile message "
+                    + "that is transmitted over the stream so that the receiver can determine when one message ends and the next message begins. Users should "
+                    + "ensure that the FlowFile content does not contain the delimiter character to avoid errors. If it is not possible to define a delimiter "
+                    + "character that is not present in the FlowFile content then the user must use another processor to change the encoding of the data e.g. Base64 "
+                    + "encoding.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .expressionLanguageSupported(true)
+            .build();
+    public static final PropertyDescriptor CONNECTION_PER_FLOWFILE = new PropertyDescriptor.Builder()
```

Comment:
While I'm not sure about use-cases I don't see a problem adding it as a property which defaults to false. I assume @aviewfromspace1 had a specific use-case in mind when adding the property and since it is optional I like keeping it.

Also PutTCP is marked as "TriggerWhenEmpty" so it will properly pruneIdleSenders just like PutSplunk.